src/crimson/os/seastore/segment_manager/ephemeral.cc

```diff
@@ -148,11 +170,20 @@ SegmentManager::read_ertr::future<> EphemeralSegmentManager::read(
   size_t len,
   ceph::bufferptr &out)
 {
-  if (addr.segment >= get_num_segments())
+  if (addr.segment >= get_num_segments()) {
+    logger().error(
+      "EphemeralSegmentManager::read: invalid segment {}",
+      addr);
```

Comment:
I hit a random low-level error here and the test crushed, using
```
$ perf_staged_fltree --track=0 --backend=seastore --default-log-level=trace
```
And the same logic works good without seastore backend
```
$ perf_staged_fltree --track=0 --backend=dummy
```
So I guess there're some corner case issues. I don't know how this happens yet, but the log is very clear that it happens during `read_extents`, before getting `CachedExtent`. Would you mind to have a look? Detail is at https://github.com/ceph/ceph/pull/36986#issuecomment-709717094

Reply:
See my comment.  This PR will have fixes for it also once I fix up the other comments.